src/features/session_trace/aggregate/index.js

```diff
@@ -93,11 +94,13 @@ export class Aggregate extends AggregateBase {
     registerHandler('trace-jserror', (...args) => this.traceStorage.storeErrorAgg(...args), this.featureName, this.ee)
     registerHandler('pvtAdded', (...args) => this.traceStorage.processPVT(...args), this.featureName, this.ee)
 
-    if (typeof PerformanceNavigationTiming !== 'undefined') {
-      this.traceStorage.storeTiming(globalScope.performance?.getEntriesByType?.('navigation')[0])
-    } else {
-      this.traceStorage.storeTiming(globalScope.performance?.timing, true)
-    }
+    single(() => {
+      if (typeof PerformanceNavigationTiming !== 'undefined') {
+        this.traceStorage.storeTiming(globalScope.performance?.getEntriesByType?.('navigation')[0])
+      } else {
+        this.traceStorage.storeTiming(globalScope.performance?.timing, true)
+      }
+    })
```

Comment:
This doesn't work. Notwithstanding the fn returned by `single(...)` is never actually executed, every time `initialize()` is called, this'll simply get a single wrap of a "new" function. Make this an instance field defined above the constructor instead:
```suggestion
  collectNavTimingOnce = single(() => {
      if (typeof PerformanceNavigationTiming !== 'undefined') {
        this.traceStorage.storeTiming(globalScope.performance?.getEntriesByType?.('navigation')[0])
      } else {
        this.traceStorage.storeTiming(globalScope.performance?.timing, true)
      }
    })
```
then call it here
```suggestion
    this.collectNavTimingOnce()
```

Reply:
Example of what I mean:
<img width="367" alt="image" src="https://github.com/user-attachments/assets/b76e8e20-bd32-4b32-897b-4769521327a2">

Reply:
ty